pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+    def _choose_and_init_cluster_environment(self) -> None:
+        self.cluster_environment: ClusterEnvironment = LightningEnvironment()
+        if isinstance(self._cluster_environment_flag, ClusterEnvironment):
+            self.cluster_environment = self._cluster_environment_flag
+        elif self._is_slurm_managing_tasks():
+            rank_zero_info("Multiprocessing is handled by SLURM.")
+            self.cluster_environment = SLURMEnvironment()
+        else:
+            for env_type in (BaguaEnvironment, TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment):
+                if env_type.detect():
+                    self.cluster_environment = env_type()
 
     @property
     def _is_sharded_training_type(self) -> bool:
         return isinstance(self._strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy))
 
-    @property
-    def _is_fully_sharded_training_type(self) -> bool:
-        return isinstance(self._strategy, DDPFullyShardedStrategy)
-
-    @property
-    def is_distributed(self) -> bool:
-        # Used for custom plugins.
-        # Custom plugins should implement is_distributed property.
-        if hasattr(self.strategy, "is_distributed") and not self.use_tpu:
-            return self.strategy.is_distributed
-        is_distributed = self.use_ddp or self.use_ddp2 or self.use_horovod
-        if self.use_tpu:
-            is_distributed |= self.strategy.is_distributed
-        return is_distributed
-
-    @property
-    def num_gpus(self) -> int:
-        gpus = self.parallel_device_ids
-        if gpus is None:
-            return 0
-        return len(gpus)
+    def _is_slurm_managing_tasks(self):
+        """used by choosing cluster enviroment."""
+        if not SLURMEnvironment.detect() or SLURMEnvironment.job_name() == "bash":
+            return False
 
-    @property
-    def num_ipus(self) -> int:
-        if isinstance(self.ipus, int):
-            return self.ipus
-        if isinstance(self._strategy, IPUStrategy):
-            return self._strategy.replication_factor
-        return 0
+        total_requested_devices = len(self._parallel_devices) * self._num_nodes_flag
+        num_slurm_tasks = int(os.environ["SLURM_NTASKS"], 0)
+        return num_slurm_tasks == total_requested_devices
 
-    @property
-    def parallel_devices(self) -> List[Union[torch.device, int]]:
-        if self.use_gpu:
-            devices = [torch.device("cuda", i) for i in self.parallel_device_ids]
-        elif self.use_tpu:
-            # explicitly don't make a tpu device here!
-            # https://github.com/PyTorchLightning/pytorch-lightning/issues/3169
-            if isinstance(self.tpu_cores, int):
-                devices = list(range(self.tpu_cores))
-        elif self.use_ipu:
-            devices = list(range(self.num_ipus))
+    def _choose_strategy(self) -> None:
+        if self._accelerator_flag == "ipu":
+            self._strategy_flag = "ipu_strategy"
```

Comment:
```suggestion
            self._strategy_flag = IPUStrategy.strategy_name
```

Let's use the strategy name. Those would be hard to maintain otherwise.

Same above.

Reply:
will update them all